centreon/www/front_src/src/ResourceAccessManagement/Listing/Listing.tsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from 'react';
+
+import { prop } from 'ramda';
+import { useSetAtom, useAtom } from 'jotai';
+
+import { MemoizedListing } from '@centreon/ui';
+
+import {
+  limitAtom,
+  pageAtom,
+  resourceAccessRulesNamesAtom,
+  selectedRowsAtom,
+  sortFieldAtom,
+  sortOrderAtom
+} from '../atom';
+import { ResourceAccessRuleType } from '../models';
+
+import useListingColumns from './columns';
+import useLoadResourceAccessRules from './useLoadResourceAccessRules';
+
+const ResourceAccessRulesListing = (): JSX.Element => {
+  const columns = useListingColumns();
+
+  const [selectedColumnIds, setSelectedColumnIds] = useState(
+    columns.map(prop('id'))
+  );
+
+  const resetColumns = (): void => {
+    setSelectedColumnIds(columns.map(prop('id')));
+  };
+
+  const [selectedRows, setSelectedRows] = useAtom(selectedRowsAtom);
+  const [sortO, setSortO] = useAtom(sortOrderAtom);
+  const [sortF, setSortF] = useAtom(sortFieldAtom);
+  const [page, setPage] = useAtom(pageAtom);
+  const setResourceAccessRulesNames = useSetAtom(resourceAccessRulesNamesAtom);
+  const setLimit = useSetAtom(limitAtom);
+  const { data: listingData, loading, refetch } = useLoadResourceAccessRules();
+
+  useEffect(() => {
+    refetch();
+  }, []);
+
+  useEffect(() => {
+    if (listingData) {
+      const names = listingData.result.map((item) => ({
+        id: item.id,
+        name: item.name
+      }));
+      setResourceAccessRulesNames(names);
+    }
+  }, [listingData]);
+
+  const changeSort = ({ sortOrder, sortField }): void => {
+    setSortF(sortField);
+    setSortO(sortOrder);
+  };
+
+  const changePage = (updatedPage: number): void => {
+    setPage(updatedPage + 1);
+  };
+
+  const predefinedRowsSelection = [
+    {
+      label: 'activated',
+      rowCondition: (row: ResourceAccessRuleType): boolean => row.isActivated
+    },
+    {
+      label: 'deactivated',
+      rowCondition: (row: ResourceAccessRuleType): boolean => !row?.isActivated
+    }
+  ];
```

Comment:
As you have some logic, it's better to extract it into a custom hook such as `useListing`

Reply:
Done